### PR TITLE
Set a default value to the registry

### PR DESCRIPTION
**Pull request by achntrl**
### What does this PR do?

Sets a default path to the registry

### Motivation

There is no default path for the registry. It is created in the folder from where the agent is started. The registry might not be accessible to the agent

### Additional Notes

How do we ensure that a client who upgrade the agent on windows migrate his current registry file ? If we don't, the client might end up with duplicated logs

### CODE REVIEW

**Comment by ajacquemot** [comment]
How do you explain that some users get their log replayed at restart on windows ?

File: pkg/config/config_windows.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ package config
 const (
 	defaultConfdPath            = "c:\\programdata\\datadog\\conf.d"
 	defaultAdditionalChecksPath = "c:\\programdata\\datadog\\checks.d"
-	defaultRunPath              = ""
+	defaultRunPath              = "c:\\programdata\\datadog"
```
Comment (ajacquemot) [inline pkg/config/config_windows.go]: nits: `c:\\programdata\\datadog\run`

**Review by ajacquemot — DISMISSED** [review]
LGTM, I am just wondering why do people see all their logs replayed.

File: releasenotes/notes/add-default-value-to-registry-3a91e6b20f29f428.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    "Add a default value to the registry on Windows: C:\ProgramData\datadog"
```
Comment (hush-hush) [inline releasenotes/notes/add-default-value-to-registry-3a91e6b20f29f428.yaml]: The release notes are meant for the users. We should explain the impact for them (ie: why we did this).

Comment (NBParis) [reply releasenotes/notes/add-default-value-to-registry-3a91e6b20f29f428.yaml]: Suggested:

`Adding a default location for the file storing pointers to make sure we never loose any logs.`

File: pkg/config/config_windows.go [hunk 3]
```diff
@@ -8,7 +8,7 @@ package config
 const (
 	defaultConfdPath            = "c:\\programdata\\datadog\\conf.d"
 	defaultAdditionalChecksPath = "c:\\programdata\\datadog\\checks.d"
-	defaultRunPath              = ""
+	defaultRunPath              = "c:\\programdata\\datadog\\run"
```
Comment (derekwbrown) [inline pkg/config/config_windows.go]: The alternative would be to not make changes to the install at all, and just put it at the root
`c:\\programdata\\datadog` would be fine.  But that's up to you

File: omnibus/resources/agent/msi/source.wxs.erb [hunk 4]
```diff
@@ -344,6 +344,16 @@
...
+      <Directory Id="run" Name="run">
+        <Component Id="run"
+                   Guid="5acf7ec1-3bcf-40d7-8840-f8de8a472c3c"
+                   Permanent="yes"
+                   Location="either"
+                   NeverOverwrite="yes">
+          <CreateFolder/>
+        </Component>
+      </Directory>
+
```
Comment (derekwbrown) [inline omnibus/resources/agent/msi/source.wxs.erb]: Argh.  Github won't let me put in a marker where you need this.  Below, line ~434, you need an entry to actually create this directory like 
`<ComponentRef Id="run" />`

Also, a more descriptive name/id would be nice, and/or a comment as to what this is for.

File: pkg/config/config_windows.go (hunk 3 above)
Comment (derekwbrown) [inline pkg/config/config_windows.go]: The alternative would be to not make changes to the install at all, and just put it at the root
c:\\programdata\\datadog would be fine. But that's up to you

**Review by derekwbrown — CHANGES_REQUESTED** [review]

**Review by ajacquemot — APPROVED** [review]

**Review by derekwbrown — APPROVED** [review]

**Pull request merged**